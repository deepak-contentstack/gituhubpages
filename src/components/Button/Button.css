@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.ui-component button{
  	margin: 0 calc(var(--space-2) + 1px);
}

/* Button */

.Button {
	background-color: transparent;
	border: 1px solid transparent;
	border-radius: $radii-6;
	font-family: $font-family-primary;
	font-size: $size-font-medium;
	font-weight: $font-weight-bold;
	line-height: $space-12;
	/* line-height: calc(var(--space-12) + 1px); */
	min-height: $space-32;
	min-width: $space-32;
	padding: $space-8 $space-16;
	/* padding: $space-8 $space-16 calc(var(--space-8) + 1px); */
	position: relative;
	text-align: center;
	transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
	border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
	vertical-align: middle;

	&__loader{
		left: 50%;
		position: absolute;
		top: 50%;
		-webkit-transform: translateX(-50%) translateY(-50%);
		-moz-transform: translateX(-50%) translateY(-50%);
	}

	&__icon {
		align-items: center;
		color: $color-base-gray-20;
		display: flex;
		height: 100%;
		justify-content: center;
		left: 0;
		position: absolute;
		top: 0;
		transition: border-color 0.15s ease-in-out;
		width: 2rem;
	}

	&.Button--icon {
		padding: 0;
		svg {
			margin : auto!important;
		}
	}

	&.Button--icon-alignment-right{
		svg:first-child{
		float: right;
		margin-left: 0.625rem;
		margin-right: 0;
		}
	}

	&:focus {
		box-shadow: 0px 0px 0px 4px #BBB4F4;
		outline: none;
	}

	&--state {
		&-disabled {
			background-color: $color-brand-secondary-lighter !important;
			border: 1px solid transparent;
			color: $color-font-tertiary !important;
			opacity: $opacity-5;
			pointer-events: none;
			text-shadow: none !important;
			& svg path {
				fill: $color-font-tertiary !important
			}
		}
		&-loading{
			cursor: default;
			pointer-events: none;
		}
	}

	&--style {
		&-circle {
			border-radius: $radii-50;
			padding: 0;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&--primary {
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		text-shadow: 0 1px 0 rgba(0, 0, 0, 0.16);

		&:hover {
			background-color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			background-color: $color-brand-primary-dark;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}
	}

	&--secondary {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		
		&:hover{
			border:1px solid $color-base-purple-25;
			color:$color-base-purple-25;
		}

		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		}

		&:hover {
			border : 1px solid $color-base-purple-25;
			color: $color-brand-primary-darker;
		}

		&:focus {
			box-shadow: 0px 0px 0px 4px #BBB4F4;
		}

		&:active {
			border: 1px solid $color-brand-primary-dark;
			color: $color-brand-primary-dark;
		}
	}

	&--tertiary {
		color: $color-font-link;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill: $color-font-link;
			}
		}
	}

	&--tertiary-outline {
		background: $color-brand-white-base;
		border: .0625rem solid $color-base-gray-5;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		height: 2.5rem;
		padding: .4375rem .625rem;

		&:hover {
			border: .0625rem solid $color-border-link;
			color: $color-font-link;

			svg:first-child {
				path, circle, rect {
					fill: $color-font-link;
					stroke: $color-font-link;
				}
			}
		}
		
		svg:first-child {
			margin: 0 $space-10 0 0;
			vertical-align: top;
		} 
	}

	&--outline {
		background: $color-brand-white-base;
		border: 1px solid $color-border-dark;
		color: $color-font-base;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;

		&:hover {
			border: 1px solid $color-border-link;
			color: $color-font-link;
		}
	}

	&--success {
		background-color: $color-brand-white-base;
		color: $color-font-success;
		&:hover {
			background-color: $color-brand-success-base;
			border: 1px solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--control {
		border: 1px solid $color-base-gray-5;
		border-radius: $radii-8;
		color: $color-base-gray-20;
		font-size: $size-font-large;
		font-weight: $font-weight-regular;
		line-height: 1;
		max-width: 11.563rem;
		overflow: hidden;
		padding: $space-12 $space-12 $space-12 $space-30;
		position: relative;
		text-overflow: ellipsis;
		white-space: nowrap;

		&:hover {
			border-color: $color-brand-primary-base;
			color: $color-brand-primary-base;
			
			svg {
				path {
				stroke: $color-brand-primary-base;
				}
			}
		}
	}

	&--delete {
		border: 1px solid $color-border-warning;
		color: $color-font-warning;
		svg {
			margin: 0 $space-10 0 0;
			vertical-align: top;
			path {
				fill : $color-base-orange-5;
			}
		}
		&:hover {
			background-color: $color-brand-warning-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-delete {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-link;
		color: $color-font-link;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-base-purple-5;
			}
		}
		&:hover{
			background-color:$color-brand-warning-base;
			border: 1px solid $color-border-warning;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
				}
			}
		}
	}

	&--outline-success {
		background-color: $color-brand-white-base;
		border: 1px solid $color-border-success;
		color: $color-font-success;
		svg {
			margin: 0 .625rem 0 0;
			vertical-align: bottom;
		}
		&:hover {
			background-color: $color-brand-success-base;
			color: $color-font-white;
			svg {
				path {
					fill : $color-base-white-5;
					stroke: $color-base-white-5;
				}
			}
		}
	}
	
	&--bulk-delete{
		background-color: $color-brand-primary-base;
		color: $color-font-white;
		svg {
			margin: 0 .625rem 0 0;
			path {
				fill : $color-font-white;
			}
		}
		&:hover{
			background-color: $color-font-white;
			color: $color-base-orange-5;
			svg {
				path {
					fill : $color-base-orange-5;
				}
			}
		}
	}

	&--danger {
		background-color: $color-brand-white-base;
		color: $color-font-warning;
		&:hover {
			background-color: $color-brand-warning-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}
	}

	&--warning {
		background-color: $color-brand-white-base;
		color: $color-font-attention;
		&:hover {
			background-color: $color-brand-attention-base;
			border: 0.0625rem solid $color-base-white-5;
			color: $color-font-white;
		}

		&-link {
			color: $color-base-purple-5;
			font-weight: $font-weight-regular;
			text-decoration: none;
		}
	}

	&--light {
		color: $color-brand-secondary-base;
		&:hover {
			background-color: $color-brand-secondary-lighter;
			color: $color-font-base;
		}
	}

	&--white {
		background: $color-brand-white-base;
		box-shadow: 0 8px 8px rgba(0, 0, 0, 0.02), 0 4px 8px rgba(0, 0, 0, 0.06), 0 8px 16px rgba(0, 0, 0, 0.04);
		color: $color-font-secondary;
		font-weight: $font-weight-regular;
		svg {
			margin-right: $space-10;
		}
		&:hover {
			background: $color-brand-secondary-lighter;
			color: $color-font-black;
			svg path {
				fill : $color-base-gray-20;
			}
		}
	}

	&--size {
		&-large {
			font-size: $size-font-xl;
			padding: $space-12 $space-18 $space-12;
		}

		&-small {
			font-size: $size-font-medium;
			min-height: $space-25;
			padding: $space-5 $space-10;
		}

		&-fullwidth {
			width: 100%;
		}
	}

	&:active {
		border: 1px solid $color-brand-primary-dark;
	}

	.badge {
		margin-left: $space-10;

		&:first-child {
			margin: 0;
			margin-right: $space-10;
		}
	}
}

[type='button']:not(:disabled),
[type='reset']:not(:disabled),
[type='submit']:not(:disabled),
button:not(:disabled) {
  	cursor: pointer;
}

.Button-wrapper button {
  	margin: 0 calc(var(--space-2) + 1px);
}

.Button-container {
	height: 6.25rem;
	margin: $space-15;
	padding: $space-34 0;
	text-align: center;
	width: 12.5rem;
}
