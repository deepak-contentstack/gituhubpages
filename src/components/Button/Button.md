# Button

## Best practices

- Use clear and accurate labels.
- Start button labels with verbs, for example "Add field", not "New field".
- Keep the position of the buttons consistent on the interface.
- Reduce the complexity by using a small number of actions on the interface. It may create confusion if too many actions are used.
