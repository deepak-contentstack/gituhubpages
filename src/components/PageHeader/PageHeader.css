@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/variable.css';
@import '../../static/styles/css/common.css';

.PageHeader {
    align-items: center;
    display: flex;
    font-family: $font-family-primary;
    justify-content: space-between;
    position: relative;
    width: 100%;

    .PageTitleWrapper {
        align-items: center;
        display:flex;

        .PageHeaderIcon {
            align-items: center;
            cursor: pointer;
            display: block;
            display: flex;
            height: calc(var(--space-22) + var(--space-5));
            padding-left: $space-6;
        }
    }

    &--Sidebar{
        margin-left: 2.813rem;
        .SidebarWindow {
            height: 100vh;
            padding-bottom: 1.75rem;
            right: -1.125rem;
            top: -1.75rem;

            &__tab-item {
                &:after {
                    display: none;
                }
            }
            &__content {
                &__title {
                    padding-top: $space-14;
                }
            }
            &__tabs-container {
                margin-top: $space-14;
            }
        }
    }
}

.PageSubTitle {
    color: $color-font-base;
    font-size: $size-font-small;
    font-weight: $font-weight-medium;
    position: absolute;
    top: $space-36;
}
