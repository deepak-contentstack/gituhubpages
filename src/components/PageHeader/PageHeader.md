## **PageHeader**

A page header consists of title, title content, information, and set of CTA's called as actions.

#### **Neccessary Props**:

- **title**: Object

  - required
  - Structure: { label : string , component :JSX | React Node , info : url }

- **content**: String or React component

  - optional

- **actions**: Array

  - optional
  - Structure: [ { label : string , onClick : MouseEvent , type : string } ]
