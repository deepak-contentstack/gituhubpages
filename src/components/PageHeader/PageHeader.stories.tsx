import React from "react";
import { withA11y } from "@storybook/addon-a11y";
import { withKnobs, object, text } from "@storybook/addon-knobs";
import notes from "./PageHeader.md";
import { PageHeader, Button, Icon, Dropdown } from "../../index";
import SideBarWindow from '../SideBarWindow/SideBarWindow';

export const Default = () => {
  const pageActions: any = [
    {
      label: 'Cancel',
      onClick: () => alert('Cancel button clicked'),
      type: 'primary',
    },
  ]
  return (
    <PageHeader title={object('title', { label: 'Page Title 1' })} actions={pageActions} />
  )
}

export const EditableTitle = () => {
  const pageActions: any = [
    {
      label: 'Cancel',
      onClick: () => alert('Cancel button clicked'),
      type: 'primary',
    },
  ]
  return (
    <PageHeader title={object('title', { label: 'Page Title 1', editable: true })} actions={pageActions} />
  )
}

export const HeaderWithInformation = () => {
  const pageActions: any = [
    {
      label: "Cancel",
      onClick: () => alert("Cancel button clicked"),
      type: "primary",
    },
  ];
  return (
    <PageHeader
      title={object("title", { label: "Page Title 1", info: "#" })}
      actions={pageActions}
    />
  );
};

export const MultipleActions = () => {
  const pageActions: any = [
    {
      label: (
        <Button buttonType="tertiary" className="ml-10" onClick={() => alert('Button clicked')}>
          Tertiary
        </Button>
      ),
    },
    {
      label: (
        <Button
          buttonType="secondary"
          className="ml-10"
          onClick={() => alert('Button clicked')}
        >
          Secondary
        </Button>
      ),
    },
    {
      label: (
        <Button
          buttonType="primary"
          className="ml-10"
          onClick={() => alert('Button clicked')}
        >
          Primary
        </Button>
      ),
    },
  ]
  return (
    <PageHeader title={object('title', { label: 'Page Title 1' })} actions={pageActions} />
  )
}

export const WithSidebarWindow = () => {
  const tabsInfo = [
    {
      id: 'tab1',
      tabIcon: <Icon icon="Information" />,
      tabLabel: 'Images',
      windowTitle: 'Entry Information',
      data: (
        <div>
          Style Dictionary is a build system that allows you to define styles once, in a way for
          any platform or language to consume. A single place to create and edit your styles, and
          a single command exports these rules to all the places you need them - iOS, Android,
          CSS, JS, HTML, sketch files, style documentation, or anything you can think of. It is
          available as a CLI through npm, but can also be used like any normal node module if you
          want to extend its functionality
        </div>
      ),
    },
  ];

  const pageActions: any = [
    {
      label: (
        <SideBarWindow tabsInfo={tabsInfo} />
      ),
      type: 'sidebar'
    },
  ];

  return (
    <PageHeader title={object('title', { label: 'Page Title 1' })} actions={pageActions} />
  )
}


// export const HeaderContent = () => {
//   const pageActions: any = [
//     {
//       label: "Button",
//       onClick: () => alert("Cancel button clicked"),
//       type: "primary",
//     },
//   ];
//   return (
//     <PageHeader
//       title={object("title", { label: "Page Title 1" })}
//       content={text("content", "Some header content")}
//       actions={pageActions}
//     />
//   );
// };

export const CustomComponentInHeader = () => {
  const pageActions: any = [
    {
      label: 'Cancel',
      onClick: () => alert('Cancel button clicked'),
      type: 'primary',
    },
  ]
  const AnyComponent = (
    <a className="PageHeaderIcon">
      <Icon icon="Settings" size="small" />
    </a>
  )
  return (
    <PageHeader
      title={{ label: 'Page Title 1', component: AnyComponent }}
      actions={pageActions}
    />
  )
}

export const PageHeaderAsComponent = () => {

  const list: any = [
    {
      label: 'first stack',
      default: true,
    },
    {
      label: 'second stack',
    },
    {
      label: 'third stack',
    },
    {
      label: 'fourth stack',
    },
    {
      label: 'fifth stack',
    },
    {
      label: 'sixth stack',
    }
  ]

  const pageActions: any = [
    {
      label: 'Cancel',
      onClick: () => alert('Cancel button clicked'),
      type: 'primary',
    },
  ]
  const AnyComponent = (
    <a className="PageHeaderIcon">
      <Icon icon="Settings" size="small" />
    </a>
  );

  const MockComponent = () => {
    return (
      <div>
        <div style={{ display: "flex", justifyContent: "space-between" }}>
          <Dropdown
            list={list}
            type="select"
            dropDownType='quaternary'
            withSearch={false}
            headerLabel="Organization"
          />
          <div>Manage</div>
        </div>
        <PageHeader
          title={{ label: 'Page Title 1', component: AnyComponent }}
          actions={pageActions}
        />
      </div>
    )
  }

  return (
    <PageHeader
      component={<MockComponent />}
    />
  )
}

export default {
  title: 'Components / PageHeader',
  component: Default,
  decorators: [withKnobs, withA11y],
  parameters: {
    notes: { notes },
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/5j7c2ljQw9Z8WghgTwzXa2/Assets-Listing?node-id=411%3A191',
    },
  },
}
