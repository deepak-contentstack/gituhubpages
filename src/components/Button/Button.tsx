import React, {
  Component,
  CSSProperties,
  MouseEvent as ReactMouseEvent,
  MouseEventHandler,
  Fragment,
} from 'react'
import cn from 'classnames'

import withDeprecatedProp from '../../utils/hooks/depricatedPropsHoc';

// import Icon, { IconProps } from '../Icon2'
// import { iconName } from '../Icon2/constants'
// import AsyncLoader from '../AsyncLoader/AsyncLoader'

import './Button.css'

const defaultProps = {
  isFullWidth: false,
  isRounded: false,
  disabled: false,
  buttonType: 'primary',
  type: 'button',
  isLoading: false,
  testId: 'cs-button'
}

type DefaultProps = Readonly<typeof defaultProps>

export type ButtonProps = {
  /**
   * Class names to be appended to the className prop of the Button Component
   */
  className?: string
  children?: React.ReactNode
  /**
   * Pass the icon that want to add in button
   */
  icon?: any
  /**
   * If you want to show only icon, set the param to ‘true’
   */
  onlyIcon?: boolean
  /**
   * Determines the alignment of the icon
   */
  iconAlignment?: 'left' | 'right'
  /**
   * Pass additional props for icon
   */
  // iconProps?: Partial<IconProps>
  /**
   * Pass an id to the button
   */
  id?: string
  /**
   * Determine the type of button you want to use
   */
  buttonType?:
  | 'primary'
  | 'secondary'
  | 'tertiary'
  | 'tertiary-outline'
  | 'outline'
  | 'success'
  | 'danger'
  | 'warning'
  | 'light'
  | 'control'
  | 'delete'
  | 'bulk-delete'
  | 'outline-delete'
  | 'outline-success'
  | 'white'
  /**
   * Determine the size of the button: small or large
   */
  size?: 'small' | 'large'
  /**
   * If button covers full width
   */
  isFullWidth?: boolean
  /**
   * If button is rounded
   */
  isRounded?: boolean
  /**
   * An ID used for testing purposes applied as a data attribute (data-test-id)
   */
  testId?: string
  /**
   * Used to set the button in the loading state
   */
  isLoading?: boolean
  /**
   * Determines the color of the loading animation you want to display
   */
  loadingColor?: string
  /**
   * Action that you want to perform when a button is clicked
   */
  onClick?: MouseEventHandler
  /**
   * Determines whether button is disabled or not
   */
  disabled?: boolean
  /**
   * If you want a button as a link, provide the details in href.
   */
  href?: string
  /**
   * Type of button action (button, submit,reset)
   */
  type?: 'button' | 'submit' | 'reset'
  /**
   * Pass any CSS properties for the button
   */
  style?: CSSProperties
} & Partial<DefaultProps>
export class Button extends Component<ButtonProps> {
  public static propTypes: any = {}
  static defaultProps = defaultProps
  render() {
    const {
      className,
      children,
      buttonType,
      size,
      isFullWidth,
      isRounded,
      onClick,
      disabled,
      href,
      type,
      icon,
      iconAlignment,
      style,
      isLoading,
      loadingColor,
      onlyIcon,
      testId,
      id,
      iconProps = {},
      ...otherProps
    }: any = this.props
    const classNames = cn('Button', className, [`Button--${buttonType}`], {
      [`Button--state-disabled`]: disabled,
      [`Button--size-${size}`]: size,
      ['Button--style-fullwidth']: isFullWidth,
      ['Button--style-circle']: isRounded,
      [`Button--icon-alignment-${iconAlignment}`]: iconAlignment,
      [`Button--state-loading`]: isLoading,
      [`Button--icon`]: onlyIcon,
    })
    const addButtonIconClassNames = cn('Button__icon')
    const Element = href ? 'a' : 'button'
    return (
      <Element
        data-test-id={testId}
        className={classNames}
        disabled={disabled}
        href={!disabled ? href : null}
        id={id}
        aria-label={id || 'aria-button'}
        type={type}
        style={style}
        onClick={(e: any) => {
          if (onClick) {
            onClick(e)
          }
        }}
        {...otherProps}
      >
        <div className="flex-center">
          {isLoading &&
            <div className="Button__loader">
              {/* <AsyncLoader color={loadingColor} /> */}
            </div>
          }
          <div
            style={{ visibility: !isLoading ? 'visible' : 'hidden' }}
            className={iconAlignment === "right" ? "flex-right" : "flex-v-center"}
          >
            {buttonType === 'control' && (
              <span className={addButtonIconClassNames}>
                {/* <Icon icon={iconName.PlusSign} /> */}
              </span>
            )}
            {/* {icon && <Icon icon={icon} {...iconProps} />} */}
            {children && <>{children}</>}
          </div>
        </div>
      </Element>
    )
  }
}

export default withDeprecatedProp(Button, { 'loading': 'isLoading' })