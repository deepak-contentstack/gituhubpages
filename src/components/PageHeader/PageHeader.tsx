import React, { MouseEventHandler } from 'react'
import Button from '../Button/Button'
import Icon from '../Icon2/Icon'
import PageTitle from '../PageTitle/PageTitle'
import cn from 'classnames'

// import Grid from '../Grid/Grid'
import './PageHeader.css'

interface IInfo {
  link: string
}
interface IHeaderTitle {
  label: string | React.ReactNode
  component?: React.ReactNode
  info?: string
  editable?: boolean
}

interface IHeaderContent {
  content: string | React.ReactNode
}

interface IHeaderAction {
  label: string | React.ReactNode
  onClick?: MouseEventHandler
  type?:
  | 'primary'
  | 'secondary'
  | 'success'
  | 'danger'
  | 'warning'
  | 'muted'
  | 'light'
  | 'dark'
  | 'link'
  | 'sidebar'
}

interface IPageHeader {
  title?: IHeaderTitle
  content?: string | React.ReactNode
  actions?: Array<IHeaderAction>
  className?: string
  component?: React.ReactNode
}

const Info: React.FunctionComponent<IInfo> = props => {
  const { link } = props
  return (
    <a href={link} className="PageHeaderIcon">
      <Icon icon="HelpDocs" />
    </a>
  )
}

const HeaderTitle: React.FunctionComponent<IHeaderTitle> = props => {
  const { label, component, info, editable } = props
  return (
    <div className="PageTitleWrapper">
      {label && <PageTitle className={(editable) ? 'PageTitle--editable' : ''} title={label} />}
      {info && <Info link={info} />}
      {component}
    </div>
  )
}

const HeaderContent: React.FunctionComponent<IHeaderContent> = props => {
  const { content } = props
  return <div className="PageSubTitle">{content}</div>
}

const HeaderAction: React.FunctionComponent<IHeaderAction> = props => {
  const { label, onClick, type } = props
  if (typeof label === 'string' && type) {
    return (
      <Button buttonType={type as any} onClick={onClick}>
        {label}
      </Button>
    )
  }
  else if (type === "sidebar") {
    return (
      <div className="PageHeader--Sidebar" >
        {label}
      </div>
    );
  }
  else {
    return <>{label}</>
  }
}

const PageHeader: React.FunctionComponent<IPageHeader> = props => {
  const { title, content, actions, className, component } = props
  const classNames = cn('PageHeader', className)

  return (
    <>
      {
        component ? <>{component}</> :
          <div className={classNames}>
            {/* <div className="flex-justify flex-v-center"> */}
            <HeaderTitle {...title} />
            {Array.isArray(actions) && (
              <div className="actions flex-v-center">
                {actions.map((action: IHeaderAction, index: number) => {
                  return <HeaderAction key={index} {...action} />
                })}
              </div>
            )}
            {/* </div> */}
            {content && <HeaderContent content={content} />}
          </div>
        /* <Grid rows={1} columns={2} justifyContent={'space-between'}>
          <HeaderTitle {...title} />
          {Array.isArray(actions) && (
            <div className="actions flex-v-center">
              {actions.map((action: IHeaderAction, index: number) => {
                return <HeaderAction key={index} {...action} />
              })}
            </div>
          )}
        </Grid> */
      }
    </>
  )
}

export default PageHeader
