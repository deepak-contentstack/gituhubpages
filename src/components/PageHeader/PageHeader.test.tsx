import React from 'react'
import renderer from 'react-test-renderer'

import PageHeader from './PageHeader'
import Button from '../Button/Button'
// import Icon from '../Icon/Icon'

describe('PageHeader', () => {
  it('render PageHeader with actions', () => {
    const DefaultHeader = () => {
      const pageActions = [
        {
          label: 'page label',
          onClick: () => alert('clicked'),
        },
      ]
      return <PageHeader title={{ label: 'Page header title' }} actions={pageActions} />
    }
    const tree = renderer.create(<DefaultHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('render PageHeader with information', () => {
  //   const HeaderWithInformation = () => {
  //     const pageActions = [
  //       {
  //         label: 'Cancel',
  //         onClick: () => alert('Cancel button clicked'),
  //         type: 'primary' as const,
  //       },
  //     ]
  //     return <PageHeader title={{ label: 'Page header title', info: '#' }} actions={pageActions} />
  //   }
  //   const tree = renderer.create(<HeaderWithInformation />).toJSON()
  //   expect(tree).toMatchSnapshot()
  // })

  it('render PageHeader with multiple actions', () => {
    const MultipleActions = () => {
      const pageActions = [
        {
          label: 'Cancel',
          onClick: () => alert('Cancel button clicked'),
          type: 'muted' as const,
        },
        {
          label: 'Save',
          onClick: () => alert('Save button clicked'),
          type: 'success' as const,
        },
        {
          label: (
            <Button buttonType="primary" onClick={() => alert('Button clicked')}>
              Button
            </Button>
          ),
        },
      ]
      return <PageHeader title={{ label: 'Page header title' }} actions={pageActions} />
    }
    const tree = renderer.create(<MultipleActions />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // it('render PageHeader with header content', () => {
  //   const HeaderContent = () => {
  //     const pageActions = [
  //       {
  //         label: 'Button',
  //         onClick: () => alert('Cancel button clicked'),
  //         type: 'primary' as const,
  //       },
  //     ]
  //     return (
  //       <PageHeader
  //         title={{ label: 'Page header title' }}
  //         content="Some header content"
  //         actions={pageActions}
  //       />
  //     )
  //   }
  //   const tree = renderer.create(<HeaderContent />).toJSON()
  //   expect(tree).toMatchSnapshot()
  // })

  it('render PageHeader with custom component in header', () => {
    const CustomComponentInHeader = () => {
      const pageActions = [
        {
          label: 'Cancel',
          onClick: () => alert('Cancel button clicked'),
          type: 'primary' as const,
        },
      ]
      const AnyComponent = <div></div>
      return (
        <PageHeader
          title={{ label: 'Page header title', component: AnyComponent }}
          actions={pageActions}
        />
      )
    }
    const tree = renderer.create(<CustomComponentInHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
