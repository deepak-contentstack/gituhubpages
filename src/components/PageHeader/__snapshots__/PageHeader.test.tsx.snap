// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader render PageHeader with actions 1`] = `
<div
  className="PageHeader"
>
  <div
    className="PageTitleWrapper"
  >
    <div
      className="PageTitle"
    >
      Page header title
    </div>
  </div>
  <div
    className="actions flex-v-center"
  >
    page label
  </div>
</div>
`;

exports[`PageHeader render PageHeader with custom component in header 1`] = `
<div
  className="PageHeader"
>
  <div
    className="PageTitleWrapper"
  >
    <div
      className="PageTitle"
    >
      Page header title
    </div>
    <div />
  </div>
  <div
    className="actions flex-v-center"
  >
    <button
      aria-label="aria-button"
      className="Button Button--primary"
      data-test-id="cs-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="flex-center"
      >
        <div
          className="flex-v-center"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Cancel
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`PageHeader render PageHeader with multiple actions 1`] = `
<div
  className="PageHeader"
>
  <div
    className="PageTitleWrapper"
  >
    <div
      className="PageTitle"
    >
      Page header title
    </div>
  </div>
  <div
    className="actions flex-v-center"
  >
    <button
      aria-label="aria-button"
      className="Button Button--muted"
      data-test-id="cs-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="flex-center"
      >
        <div
          className="flex-v-center"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Cancel
        </div>
      </div>
    </button>
    <button
      aria-label="aria-button"
      className="Button Button--success"
      data-test-id="cs-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="flex-center"
      >
        <div
          className="flex-v-center"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Save
        </div>
      </div>
    </button>
    <button
      aria-label="aria-button"
      className="Button Button--primary"
      data-test-id="cs-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <div
        className="flex-center"
      >
        <div
          className="flex-v-center"
          style={
            Object {
              "visibility": "visible",
            }
          }
        >
          Button
        </div>
      </div>
    </button>
  </div>
</div>
`;
